### PR TITLE
[Added] Sortable BREAD Items

### LINKED ISSUE #994: New Feature Suggestion - Order Fields
- Laravel Version: 5.4.
- Voyager Version: 0.11.7
- PHP Version: 7.0.15 & 7.1.1
- Database Driver & Version: MySQL 5.6.35 & MariaDB

### Description:
First of all thank you for creating one of best CRUD system I've used for Laravel. Respect for the people who build it!! 
One of the things I would suggest **is to order fields, just like the menu items**. 

Or some new param we could use in the details json, and use it for the order. 
```
{
  "order": "1"
}
```

Until this feature is available what I did was add a new column to this table_rows  (order) ,  and edit the query to return the fields order by **order**. But as you can imagine it's not a solution...

Regards

### PR DESCRIPTION
**Pull request by akazorg**
Making BREAD items sortable, this changes fields order on views Browse/Edit/Add.

Fixes #994.

### CODE REVIEW

File: resources/views/tools/database/edit-add-bread.blade.php [hunk 1]
```diff
@@ -270,6 +275,23 @@
                 });
             @endif
 
+            /**
+             * Reorder items
+             */
+            $('#bread-items').sortable({
```
Comment (akazorg) [inline resources/views/tools/database/edit-add-bread.blade.php:281]: I ended up using jQuery, hope it's ok.

Ping @abdgad @marktopper

**Comment by Rostom22** [comment]
How can i change the order of my fields in the edit-add view please ?

**Comment by akazorg** [comment]
Hi @Rostom22 when you edit a BREAD item, you can easily order fields by drag-and-drop rows.
Don't forget to hit save, because this isn't working with AJAX.
If you're using a older version of Voyager, make sure you have the latest assets / migrations.

Good luck!

**Comment by Rostom22** [comment]
Hi, i tried to drag-and-drop but it'sn't working, also in te edit-add view the field appears in different order. thanks for your help.

**Comment by akazorg** [comment]
Hello, when you say it doesn't work, the drag-and-drop UI effect is visible/working?
It seams you're not using the latest version.

Please check if you:
- published Voyager views before, if so, you need to make sure you update them.
- run the migration, if you did, you should see a new field named `order` in `data_types` table.

I suggest you to join us on Slack, it may be faster to help you.

Happy coding :)

**Comment by Rostom22** [comment]
Sorry i'm a begginer and i don't understand very well, i installed voyager with composer 5 days ago following the videos of devdojo. what's the Slack ? Thanks again !

**Comment by akazorg** [comment]
It's ok, we all started somewhere :)
Please keep in mind that you need to know the basis of Laravel, otherwise it's hard to jump on a project like Voyager.

Slack channel is were the community hangs out. You can get support from there.
https://laravel-voyager.slack.com/

File: src/Models/DataType.php [hunk 2]
```diff
@@ -141,7 +142,17 @@ public function fieldOptions()
...
 
-        $fieldOptions = SchemaManager::describeTable($table);
+        // Get ordered BREAD fields
+        $orderedFields = $this->rows()->pluck('field')->toArray();
+
+        $_fieldOptions = SchemaManager::describeTable($table)->toArray();
+
+        $fieldOptions = [];
+        $f_size = count($orderedFields);
+        for ($i = 0; $i < $f_size; $i++) {
```
Comment (DODMax) [inline src/Models/DataType.php:152]: I think we are loosing new fields here (added through DB edit but not yet in data_rows)

Comment (akazorg) [reply src/Models/DataType.php:152]: I'm not sure. Can you test that please?

Comment (DODMax) [reply src/Models/DataType.php:152]: The problem is that we iterate on $orderedFields which do not contains new fields. We should work on $_fieldOptions instead.
I've added a PR for that https://github.com/the-control-group/voyager/pull/1210

**Comment by Rostom22** [comment]
Hello, the drag-and-drop UI effect is not visible, to get the new version i just run composer update ? how can i get an invitation for slack ? thanks.

File: src/Models/DataType.php (hunk 2 above)
Comment (akazorg) [reply src/Models/DataType.php:152]: Great job. Thank you!!

**Pull request merged**